### PR TITLE
Allow assigning UUID job ids with Pulsar in MQ mode.

### PR DESCRIPTION
**Pull request by jmchilton**
Most the plumbing was already there from RESTful mode where this was done on the remote server. For MQ mode just add 'assign_ids: uuid' to the job destination/environment parameters.

The corresponding Pulsar client updates that enabled this are in the pull request galaxyproject/pulsar#208.

### CODE REVIEW

File: lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt [hunk 1]
```diff
@@ -121,7 +121,7 @@ pbr==5.4.2
 prettytable==0.7.2
 prov==1.5.1
 psutil==5.6.3
-pulsar-galaxy-lib==0.14.0.dev0
+pulsar-galaxy-lib==0.14.0.dev1
```
Comment (nsoranzo) [inline lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt:124]: Can you please sync the pin in the Pipfile?

Comment (jmchilton) [reply lib/galaxy/dependencies/pipfiles/default/pinned-requirements.txt:124]: Done... I think.

File: lib/galaxy/jobs/runners/pulsar.py [hunk 2]
```diff
@@ -837,14 +837,21 @@ def _monitor(self):
...
-        job_id = None
+        galaxy_job_id = None
         try:
-            job_id = full_status["job_id"]
-            job, job_wrapper = self.app.job_manager.job_handler.job_queue.job_pair_for_id(job_id)
+            remote_job_id = full_status["job_id"]
+            if len(remote_job_id) == 32:
+                # It is a UUID - assign_ids = uuid in destination params...
+                sa_session = self.app.model.context.current
+                # TODO: Add an index for this column.
```
Comment (bgruening) [inline lib/galaxy/jobs/runners/pulsar.py]: This comment can now we removed I think.

Comment (jmchilton) [reply lib/galaxy/jobs/runners/pulsar.py]: Yup... meant to... rebased.

**Review by bgruening — APPROVED** [review]

**Pull request merged**